src/libcore/tests/mem.rs

```diff
@@ -129,3 +129,21 @@ fn test_discriminant_send_sync() {
     is_send_sync::<Discriminant<Regular>>();
     is_send_sync::<Discriminant<NotSendSync>>();
 }
+
+#[test]
+fn test_const_forget() {
+    const _: () = forget(0i32);
+    const _: () = forget(Vec::<Vec<Box<i32>>>::new());
+
+    // Writing this function signature without const-forget
+    // triggers compiler errors:
+    // 1) That we use a non-const fn inside a const fn
+    // 2) without the forget, it complains about the destructor of Box
+    const fn const_forget_box<T>(x: Box<T>) {
+        forget(x);
+    }
+
+    // Call the forget_box at runtime,
+    // as we can't const-construct a box yet.
+    const_forget_box(Box::new(0i32));
```

Comment:
This test fails in Miri due to a memory leak: it complains that there is a (run-time!) allocation that did not get freed, and that allocation is the `Box` being created here.

It almost looks like that is deliberate? Do we really want a leaking test in the test suite? Cc @oli-obk

Reply:
I don't entirely see the point of calling this function at run-time, as it tests nothing about CTFE. Sure, you can't test it at compile-time, but testing at run-time instead makes no sense, it just tests the entirely wrong thing. Am I missing something?

Reply:
Oh woops, I'm sory, I didn't thought of this failing miri. (Why did the CI still succeed, btw? no `miri test`?)
The runtime call was more like a 'Proof' it still works at runtime. It can be removed.

Reply:
Due to CI budget constraints, running Miri on the libcore test suite [is a separate project](https://github.com/RalfJung/miri-test-libstd). That's why CI here did not fail.

> It can be removed.

I would prefer that. Can you submit a PR?